src/base/utils/string.h

```diff
@@ -37,18 +37,37 @@
 #include <Qt>
 #include <QtContainerFwd>
 
+#ifndef Q_OS_WIN
+#include <QCollator>
+#endif
+
 class QStringRef;
 
 namespace Utils::String
 {
-    QString fromDouble(double n, int precision);
-
-    int naturalCompare(const QString &left, const QString &right, const Qt::CaseSensitivity caseSensitivity);
-    template <Qt::CaseSensitivity caseSensitivity>
-    bool naturalLessThan(const QString &left, const QString &right)
+    class NaturalCompare
     {
-        return (naturalCompare(left, right, caseSensitivity) < 0);
-    }
+    public:
+        explicit NaturalCompare(const Qt::CaseSensitivity caseSensitivity = Qt::CaseSensitive);
```

Comment:
If case sensitivity is accepted unconditionally it should have effect unconditionally (below).

Reply:
Not sure what you mean.

Reply:
It has effect only on Windows, isn't it?